src/Explorer/Panes/DeleteCollectionConfirmationPaneComponent.tsx

```diff
@@ -0,0 +1,201 @@
+import * as NotificationConsoleUtils from "../../Utils/NotificationConsoleUtils";
+import * as TelemetryProcessor from "../../Shared/Telemetry/TelemetryProcessor";
+import * as React from "react";
+import { Action, ActionModifiers } from "../../Shared/Telemetry/TelemetryConstants";
+import { PanelFooterComponent } from "./PanelFooterComponent";
+import { Collection } from "../../Contracts/ViewModels";
+import { Text, TextField } from "office-ui-fabric-react";
+import { userContext } from "../../UserContext";
+import { Areas } from "../../Common/Constants";
+import { DefaultAccountExperienceType } from "../../DefaultAccountExperienceType";
+import { CassandraAPIDataClient } from "../Tables/TableDataClient";
+import { deleteCollection } from "../../Common/dataAccess/deleteCollection";
+import { getErrorMessage, getErrorStack } from "../../Common/ErrorHandlingUtils";
+import { DefaultExperienceUtility } from "../../Shared/DefaultExperienceUtility";
+import { PanelErrorComponent, PanelErrorProps } from "./PanelErrorComponent";
+import DeleteFeedback from "../../Common/DeleteFeedback";
+import Explorer from "../Explorer";
+import LoadingIndicator_3Squares from "../../../images/LoadingIndicator_3Squares.gif";
+
+export interface DeleteCollectionConfirmationPaneProps {
+  explorer: Explorer;
+  closePanel: () => void;
+  openNotificationConsole: () => void;
+}
+
+export interface DeleteCollectionConfirmationPaneStates {
+  formError: string;
+  isExecuting: boolean;
+}
+
+export class DeleteCollectionConfirmationPaneComponent extends React.Component<
+  DeleteCollectionConfirmationPaneProps,
+  DeleteCollectionConfirmationPaneStates
+> {
+  private inputCollectionName: string;
+  private deleteCollectionFeedback: string;
+
+  constructor(props: DeleteCollectionConfirmationPaneProps) {
+    super(props);
+
+    this.state = {
+      formError: "",
+      isExecuting: false,
+    };
+  }
+
+  render(): JSX.Element {
+    return (
+      <div className="panelContentContainer">
+        <PanelErrorComponent {...this.getPanelErrorProps()} />
+        {this.renderConfirmDeleteSection()}
```

Comment:
I'm not a fan of this pattern of calling a function to render more JSX. Especially in the case it is not stateful. I think these should either A) be just inlined. Or B) split out into separate components entirely. 

I think by doing the `this.render*` pattern we are getting the worst of both worlds. Its not very idiomatic to react and we end up coupling our components making them more difficult to reuse later.

Reply:
I agree. I just thought that this would make it a bit easier to read. I've moved them in-line in my latest commit.